File: 프로그래머스/lv1/42889. 실패율/실패율.kt
class Solution {
    fun solution(N: Int, stages: IntArray): IntArray {
        var answer = intArrayOf()
        var a = 0.0
        var num = stages.size.toDouble()
        var ratio : MutableMap<Int,Double> = mutableMapOf()
        for(i in 0 until N){
            a = stages.count{it==i+1}.toDouble()
            ratio.put(i+1,a/(num))
            num-=a
            if(num==0.0) num =1.0
        }
        var mapList = ratio.toList().sortedByDescending{it.second}
        for(i in mapList.iterator()) answer = answer.plus(i.first)
        return answer
    }
}

File: 프로그래머스/lv1/42889. 실패율/실패율.py
def solution(N, stages):
    d={}
    l=len(stages)
    for i in range(1,N+1):
        if l==0:d[i]=stages.count(i)/1
        else:d[i]=stages.count(i)/l
        l-=stages.count(i)
        
    answer=sorted(d.items(),key=lambda x:x[1],reverse=True)
    return [answer[i][0] for i in range(len(answer))]